app/eth2wrap/eth2wrap.go

```diff
@@ -106,6 +115,94 @@ func (m multi) Address() string {
 	return m.clients[0].Address()
 }
 
+func (m multi) SubmitBeaconCommitteeSubscriptionsV2(ctx context.Context, subscriptions []*eth2exp.BeaconCommitteeSubscription) ([]*eth2exp.BeaconCommitteeSubscriptionResponse, error) {
+	const label = "submit_beacon_committee_subscriptions_v2"
+
+	res0, err := provide(ctx, m.clients,
+		func(ctx context.Context, cl Client) ([]*eth2exp.BeaconCommitteeSubscriptionResponse, error) {
+			return cl.SubmitBeaconCommitteeSubscriptionsV2(ctx, subscriptions)
+		},
+		nil,
+	)
+	if err != nil {
+		incError(label)
+		err = errors.Wrap(err, "eth2wrap")
+	}
+
+	return res0, err
+}
+
+func NewWrapHTTP(eth2Svc eth2client.Service) (Client, error) {
+	httpCl, ok := eth2Svc.(*eth2http.Service)
+	if !ok {
+		return nil, errors.New("invalid eth2http client")
+	}
+
+	return httpWrap{httpCl}, nil
+}
+
+// httpWrap implements Client by wrapping eth2http.Service,
+// also implements experimental interfaces which are absent in eth2http.Service.
+type httpWrap struct {
+	*eth2http.Service
+}
+
+type submitBeaconCommitteeSubscriptionsV2JSON struct {
+	Data []*eth2exp.BeaconCommitteeSubscriptionResponse `json:"data"`
+}
+
+// SubmitBeaconCommitteeSubscriptionsV2 implements eth2exp.BeaconCommitteeSubscriptionsSubmitterV2.
+func (h httpWrap) SubmitBeaconCommitteeSubscriptionsV2(ctx context.Context, subscriptions []*eth2exp.BeaconCommitteeSubscription) ([]*eth2exp.BeaconCommitteeSubscriptionResponse, error) {
+	var reqBodyReader bytes.Buffer
+	if err := json.NewEncoder(&reqBodyReader).Encode(subscriptions); err != nil {
```

Comment:
move a fan of just `json.Marshal` and `json.Unmarshal`

Reply:
yeah me too but I wrote this aligning with go-eth2-client pattens, we can change this to our ways not sure

Reply:
our codebase our patterns 😉